admin/runner.py

```diff
@@ -3,20 +3,77 @@
 Tools for running commands.
 """
 import sys
-from pipes import quote
-from subprocess import check_call, CalledProcessError
+import os
 
-def run(command, **kwargs):
+from characteristic import attributes
+
+from twisted.internet.error import ConnectionDone
+from twisted.internet.endpoints import ProcessEndpoint, connectProtocol
+from twisted.internet.defer import Deferred
+
+from twisted.protocols.basic import LineOnlyReceiver
+
+
+@attributes([
+    "deferred",
+    "output",
+])
+class CommandProtocol(LineOnlyReceiver, object):
```

Comment:
Am I right in recalling from another PR that this uses `characteristic` and not `pyrsistent` because LineOnlyReceiver must be mutable?